File: backend/types/menu.go
package types

// CreateMenuRequest 创建菜单请求
type CreateMenuRequest struct {
	ParentID  uint   `json:"parent_id"`
	Name      string `json:"name" binding:"required,max=50"`
	Path      string `json:"path" binding:"max=200"`
	Component string `json:"component" binding:"max=200"`
	Icon      string `json:"icon" binding:"max=50"`
	Sort      int    `json:"sort"`
	Hidden    bool   `json:"hidden"`
	Cache     bool   `json:"cache"`
	Type      int    `json:"type" binding:"required,oneof=1 2"`
}

// UpdateMenuRequest 更新菜单请求
type UpdateMenuRequest struct {
	ParentID  uint   `json:"parent_id"`
	Name      string `json:"name" binding:"required,max=50"`
	Path      string `json:"path" binding:"max=200"`
	Component string `json:"component" binding:"max=200"`
	Icon      string `json:"icon" binding:"max=50"`
	Sort      int    `json:"sort"`
	Status    int    `json:"status" binding:"oneof=0 1"`
	Hidden    bool   `json:"hidden"`
	Cache     bool   `json:"cache"`
	Type      int    `json:"type" binding:"required,oneof=1 2"`
}

// MenuResponse 菜单响应
type MenuResponse struct {
	ID        uint           `json:"id"`
	ParentID  uint           `json:"parent_id"`
	Name      string         `json:"name"`
	Path      string         `json:"path"`
	Component string         `json:"component"`
	Icon      string         `json:"icon"`
	Sort      int            `json:"sort"`
	Status    int            `json:"status"`
	Hidden    bool           `json:"hidden"`
	Cache     bool           `json:"cache"`
	Type      int            `json:"type"`
	Children  []MenuResponse `json:"children,omitempty"`
}

// MenuListRequest 菜单列表请求
type MenuListRequest struct {
	Keyword string `form:"keyword"`
	Type    int    `form:"type"`
}

// MenuListResponse 菜单列表响应
type MenuListResponse struct {
	List []MenuResponse `json:"list"`
}

File: web/src/api/menu.ts
import request from '@/utils/request';

export interface CreateMenuRequest {
  parent_id: number;
  name: string;
  path?: string;
  component?: string;
  icon?: string;
  sort?: number;
  hidden?: number;
  cache?: number;
  type: 1 | 2;
  is_external?: number; // 是否为外部链接，0-否，1-是
  external_url?: string; // 外部链接地址
  open_type?: number; // 打开方式，0-内嵌，1-新窗口
}

export interface UpdateMenuRequest extends Partial<CreateMenuRequest> {
  status?: 0 | 1;
}

export interface MenuResponse {
  id: number;
  parent_id: number;
  name: string;
  path: string;
  component: string;
  icon: string;
  sort: number;
  status: 0 | 1;
  hidden: 0 | 1;
  cache: 0 | 1;
  type: 1 | 2;
  is_external?: number; // 是否为外部链接，0-否，1-是
  external_url?: string; // 外部链接地址
  open_type?: number; // 打开方式，0-内嵌，1-新窗口
  children?: MenuResponse[];
}

export interface MenuListRequest {
  keyword?: string;
  type?: 1 | 2;
}

export interface MenuListResponse {
  list: MenuResponse[];
}

// 获取用户菜单列表
export function getUserMenus() {
  return request.get<MenuListResponse>('/user/menus');
}

// 获取所有菜单列表（用于菜单管理）
export function getMenuList(params: MenuListRequest = {}) {
  return request.get<MenuListResponse>('/menus', { params });
}

// 创建菜单
export function createMenu(data: CreateMenuRequest) {
  return request.post('/menus', data);
}

// 更新菜单
export function updateMenu(id: number, data: UpdateMenuRequest) {
  return request.put(`/menus/${id}`, data);
}

// 删除菜单
export function deleteMenu(id: number) {
  return request.delete(`/menus/${id}`);
}
